addons/mail/controllers/discuss.py

```diff
@@ -1,18 +1,221 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import base64
-import werkzeug.utils
-import werkzeug.wrappers
+import json
 
 from odoo import http
-from odoo.exceptions import AccessError
 from odoo.http import request
+from odoo.tools import consteq
+from odoo.tools.misc import get_lang
+from odoo.tools.translate import _
+from werkzeug.exceptions import NotFound
 
 
 class DiscussController(http.Controller):
 
-    @http.route('/mail/read_followers', type='json', auth='user')
+    # --------------------------------------------------------------------------
+    # Public Pages
+    # --------------------------------------------------------------------------
+
+    @http.route('/chat/<int:channel_id>/<string:invitation_token>', methods=['GET'], type='http', auth='public')
+    def discuss_channel_invitation(self, channel_id, invitation_token, **kwargs):
+        channel_sudo = request.env['mail.channel'].browse(int(channel_id)).sudo().exists()
+        if not channel_sudo or not channel_sudo.uuid or not consteq(channel_sudo.uuid, invitation_token):
+            raise NotFound()
```

Comment:
is there no limitation on which kind of channel you can join with this? It seems this should not be possible for livechats, nor for other private channels?

Reply:
In terms of security if you have the uuid you can already sneak in and receive messages for all kind of channels before this PR (support for uuid bus notif + if livechat is actually installed there are routes to fetch all messages based on uuid).
This new route just makes people using it formally a member of said channel.

But it would make sense to at least prevent it for channel of type chat, as those are exclusively between 2 persons are not supposed to be joined in any way.

Private channels (of channel type group in particular) are the main target of this PR, the spec is to create a private group on the fly and invite people with a link.

"Public" channel (which is currently equals to visible to all internal users) is actually the least likely to be shared this way.

Reply:
As for livechat it's actually common enough for a livechat operator to request help from his colleague on some discussion, in which case being able to share the link with him is more convenient than the (already long time existing) invite feature by name.